scripts/CodeCoverage.py

```diff
@@ -65,12 +68,29 @@ def unzip_file(full_file_path, directory_to_extract_to):
     to_unzip.extractall(directory_to_extract_to)
     to_unzip.close()
 
-def run_CMake_cmd():
-    CMAKE = '/usr/local/probe/bin/cmake'
+
+def get_cmake(): 
+    if PROBE_BUILD:
+        print "return probe cmake"
+        return '/usr/local/probe/bin/cmake'
+    return 'cmake'
+
+def get_make_arguments():
+    if PROBE_BUILD:
+        return '-DCMAKE_CXX_COMPILER_ARG1:STRING=\'-std=c++14 -Wall -Werror -g -gdwarf-2 -fno-elide-constructors -fprofile-arcs -ftest-coverage -O0 -fPIC -m64 -Wl,-rpath -Wl,. -Wl,-rpath -Wl,/usr/local/probe/lib -Wl,-rpath -Wl,/usr/local/probe/lib64 -fno-inline -fno-inline-small-functions -fno-default-inline\''
+    return '-DCMAKE_CXX_COMPILER_ARG1:STRING=\'-Wall -Werror -g -gdwarf-2 -fno-elide-constructors -fprofile-arcs -ftest-coverage -O0 -fPIC -m64  -fno-inline -fno-inline-small-functions -fno-default-inline\''
```

Comment:
Since the default make arguments are a subset of the probe specific ones, you could remove the duplication with something like:

``` python
default_args = "-Wall -Werror -g -gdwarf-2 -fno-elide-constructors -fprofile-arcs -ftest-coverage -O0 -fPIC -m64  -fno-inline -fno-inline-small-functions -fno-default-inline"
if PROBE_BUILD:
   default_args += "-std=c++14 -Wl,-rpath -Wl,. -Wl,-rpath -Wl,/usr/local/probe/lib -Wl,-rpath -Wl,/usr/local/probe/lib64"
return "-DCMAKE_CXX_COMPILER_ARG1:STRING='"+default_args+"'"
```